lib/util/object_utils.js

```diff
@@ -94,6 +94,35 @@ shaka.util.ObjectUtils = class {
     return clone;
   }
 
+  /**
+   * Performs a shallow comparison between two objects.
+   *
+   * @template T
+   * @param {T} a
+   * @param {T} b
+   * @return {boolean}
+   */
+  static shallowCompare(a, b) {
+    if (Object.is(a, b)) {
+      return true;
+    }
+    if (typeof a !== 'object' || a === null ||
+        typeof b !== 'object' || b === null) {
+      return false;
+    }
+    const keysA = Object.keys(a);
+    const keysB = Object.keys(b);
+    if (keysA.length !== keysB.length) {
+      return false;
+    }
+    for (let i = 0; i < keysA.length; i++) {
+      if (!Object.is(a[keysA[i]], b[keysA[i]])) {
+        return false;
+      }
+    }
+    return true;
+  }
```

Comment:
My understanding of the shaka-player config system is that for objects it allows them to either be null (non object fields are not allowed to be null) or an object that has a subset of the possible properties, missing, undefined and invalid properties are replaced with their respective default values. Which means that this method will always receive either an object with the expected shape or null, so as there are only 4 properties it might make more sense to add explicit comparisons for said fields instead of the current generic approach.

Ref: https://github.com/shaka-project/shaka-player/blob/main/lib/util/config_utils.js#L29

Reply:
We had a bit of a discussion going on https://github.com/shaka-project/shaka-player/pull/8265 where we'd introduce two maps of `CSSStyleDeclaration` and figured we'd best shallowly compare these too in order to figure out if we need to force render captions on configure.

My main intention for introducing shallowCompare goes beyond solely comparing `shaka.extern.BoxSides`.

Personal preference but ideally we'd treat the config as immutable, which would allow us to check objects by reference, but it's not something I'd think about now (and probably needs refining / discussions).